### PR TITLE
1) Warnings reviews and some phpDoc blocks improvements in system/Tes…

### PR DESCRIPTION
**Pull request by fmertins**
…t files.

2) composer.json, added ext-json as requirement.

### CODE REVIEW

**Comment by fmertins** [comment]
Hi folks, sorry but I can't understand why CodeIgniter seems to find the Constants.php file in the public folder... (instead of application folder)

From the Travis CI:
`PHP Fatal error:  require_once(): Failed opening required '/home/travis/build/bcit-ci/CodeIgniter4/public/Config/Constants.php' (include_path='.:/home/travis/.phpenv/versions/7.1.23/share/pear') in /home/travis/build/bcit-ci/CodeIgniter4/system/bootstrap.php on line 98`

File: system/Test/CIUnitTestCase.php [hunk 1]
```diff
@@ -125,7 +126,7 @@ protected function createApplication()
 		$systemPath = realpath(__DIR__.'/../');
 
 		require_once $systemPath.'/'.$this->configPath.'/Paths.php';
-		$paths = $this->adjustPaths(new \Config\Paths());
+		$this->adjustPaths(new Paths());
```
Comment (lonnieezell) [inline system/Test/CIUnitTestCase.php]:
```suggestion
		$paths = $this->adjustPaths(new Paths());
```

Removing that return value is why you're seeing the error you're seeing on Travis.

File: system/Test/ReflectionHelper.php [hunk 2]
```diff
@@ -95,8 +89,10 @@ private static function getAccessibleRefProperty($obj, $property)
 	 * @param object|string $obj      object or class name
 	 * @param string        $property property name
 	 * @param mixed         $value    value
+	 * @return void
+	 * @throws \ReflectionException
 	 */
-	public static function setPrivateProperty($obj, $property, $value)
+	public static function setPrivateProperty($obj, string $property, $value): void
```
Comment (lonnieezell) [inline system/Test/ReflectionHelper.php]: Not a fan of enforcing void returns, even though it's semantically correct.

File: system/Test/CIUnitTestCase.php (hunk 1 above)
Comment (lonnieezell) [reply system/Test/CIUnitTestCase.php]: I've found that PHPStorm occasionally lies about random things. :)

**Comment by fmertins** [comment]
Hi, ok I did the correction, but still Travis seems to be broken with SQLite...
Do I need to do something else?

**Comment by lonnieezell** [comment]
> I understood this var is used inside the require but I'm thinking maybe we need another approach to avoid this warning?

I'll be honest, I'm not concerned at all about this warning. And don't listen to Codacy currently as we have never gone in and set it up with our rules. It's an experiment we have had a chance to get back to and decide if it's worth keeping on. 

How would you suggest fixing it? What is here works well, is unobtrusive and solved some situations users ran into in the past with reorganizing folders, etc. If you can show an elegant solution that doesn't change a bunch of things just to avoid a warning, then cool. Otherwise - I'm not bothered by it.

**Comment by fmertins** [comment]
Morning! OK thanks, I was thinking something like "if Codacy or any other tool shows me an error/X I did something wrong and probably won't be accepted". I agree with the actual code and I'll probably move forward looking other files to improve.

**Comment by lonnieezell** [comment]
Awesome. Thanks for your contributions!

**Pull request closed without merging**